include/zephyr/dt-bindings/clock/stm32_common_clocks.h

```diff
@@ -20,35 +20,58 @@
-#define STM32_MCO_CFGR_SHIFT_SHIFT  16U
-#define STM32_MCO_CFGR_MASK_MASK    0x1FU
-#define STM32_MCO_CFGR_MASK_SHIFT   22U
-#define STM32_MCO_CFGR_VAL_MASK     0x1FU
-#define STM32_MCO_CFGR_VAL_SHIFT    27U
+/** Helper macros to pack RCC clock register info in the DT */
+#define STM32_DT_RCC_REG_MASK		0xFFFFU
+#define STM32_DT_RCC_REG_SHIFT		0U
+#define STM32_DT_RCC_SHIFT_MASK		0x3FU
+#define STM32_DT_RCC_SHIFT_SHIFT	16U
+#define STM32_DT_RCC_MASK_MASK		0x1FU
+#define STM32_DT_RCC_MASK_SHIFT		22U
+#define STM32_DT_RCC_VAL_MASK		0x1FU
+#define STM32_DT_RCC_VAL_SHIFT		27U
 
 /**
- * @brief STM32 MCO configuration register bit field
+ * @brief Pack STM32 source clock selection RCC register bit fields for the DT
  *
- * @param reg    Offset to RCC register holding MCO configuration
- * @param shift    Position of field within RCC register (= field LSB's index)
- * @param mask    Mask of register field in RCC register
- * @param val    Clock configuration field value (0~0x1F)
+ * @param val    Clock configuration field value
+ * @param mask   Mask of register field in RCC register
+ * @param shift  Position of field within RCC register (= field LSB's index)
+ * @param reg    Offset to target clock configuration register in RCC
  *
- * @note 'reg' range:    0x0~0xFFFF    [ 00 : 15 ]
- * @note 'shift' range:    0~63        [ 16 : 21 ]
- * @note 'mask' range:    0x00~0x1F    [ 22 : 26 ]
+ * @note 'reg' range:    0x0~0xFFFF   [ 00 : 15 ]
+ * @note 'shift' range:  0~63         [ 16 : 21 ]
+ * @note 'mask' range:   0x00~0x1F    [ 22 : 26 ]
  * @note 'val' range:    0x00~0x1F    [ 27 : 31 ]
- *
  */
-#define STM32_MCO_CFGR(val, mask, shift, reg)                        \
-	((((reg) & STM32_MCO_CFGR_REG_MASK) << STM32_MCO_CFGR_REG_SHIFT) |        \
-	 (((shift) & STM32_MCO_CFGR_SHIFT_MASK) << STM32_MCO_CFGR_SHIFT_SHIFT) |    \
-	 (((mask) & STM32_MCO_CFGR_MASK_MASK) << STM32_MCO_CFGR_MASK_SHIFT) |        \
-	 (((val) & STM32_MCO_CFGR_VAL_MASK) << STM32_MCO_CFGR_VAL_SHIFT))
+#define STM32_DT_RCC_CLOCK_SELECT(val, mask, shift, reg)                          \
+	((((reg) & STM32_DT_RCC_REG_MASK) << STM32_DT_RCC_REG_SHIFT) |        \
+	 (((shift) & STM32_DT_RCC_SHIFT_MASK) << STM32_DT_RCC_SHIFT_SHIFT) |  \
+	 (((mask) & STM32_DT_RCC_MASK_MASK) << STM32_DT_RCC_MASK_SHIFT) |     \
+	 (((val) & STM32_DT_RCC_VAL_MASK) << STM32_DT_RCC_VAL_SHIFT))
+
+/* STM32_CLOCK_* macros, defined for convenience */
+#define STM32_CLOCK_REG_MASK		STM32_DT_RCC_REG_MASK
+#define STM32_CLOCK_REG_SHIFT		STM32_DT_RCC_REG_SHIFT
+#define STM32_CLOCK_SHIFT_MASK		STM32_DT_RCC_SHIFT_MASK
+#define STM32_CLOCK_SHIFT_SHIFT		STM32_DT_RCC_SHIFT_SHIFT
+#define STM32_CLOCK_MASK_MASK		STM32_DT_RCC_MASK_MASK
+#define STM32_CLOCK_MASK_SHIFT		STM32_DT_RCC_MASK_SHIFT
+#define STM32_CLOCK_VAL_MASK		STM32_DT_RCC_VAL_MASK
+#define STM32_CLOCK_VAL_SHIFT		STM32_DT_RCC_VAL_SHIFT
+#define STM32_DOMAIN_CLOCK(val, mask, shift, reg) \
+	STM32_DT_RCC_CLOCK_SELECT((val), (mask), (shift), (reg))
```

Comment:
Likewise, is there a point to still distinguish `STM32_DOMAIN_CLOCK` and `STM32_MCO_CFGR`?
Since we're simplifying and factorizing, I'd go the full route and use only the new `STM32_DT_RCC_CLOCK_SELECT` in every file.
@erwango What do you think?

Reply:
I would tend to agree with you.

Reply:
Fine with me.
Let's update the full range of stm32*_clocks.h files then.